spec/helpers/application_helper/buttons/middleware_standalone_server_action_spec.rb

```diff
@@ -14,6 +14,27 @@
         it { expect(subject.visible?).to be_truthy }
       end
     end
+
+    context 'when record responds to #immutable?' do
+      def create_immutable(value)
+        {'Immutable' => value.to_s }
+      end
+
+      context 'when server.properties.immutable? == true' do
+        let(:immutable_server) { FactoryGirl.create(:middleware_server, :properties => create_immutable(true)) }
+        let(:record) { immutable_server }
+        subject { described_class.new(setup_view_context_with_sandbox({}), {}, {'record' => record}, {}) }
+        it { expect(subject).to be_visible }
+      end
+
+      context 'when server.properties.immutable? == false' do
+        let(:mutable_server) { FactoryGirl.create(:middleware_server, :properties => create_immutable(false)) }
+        let(:record) { mutable_server }
+        subject { described_class.new(setup_view_context_with_sandbox({}), {}, {'record' => record}, {}) }
+        it { expect(subject).not_to be_visible }
+      end
+    end
```

Comment:
I did think about the not testing 'Hawkular' product but it was already not part of the test suite and I didn't want to add permutation explosion. Not sure how we should handle this?